packages/ember-glimmer/lib/syntax/mount.js

```diff
@@ -0,0 +1,97 @@
+import { ArgsSyntax, StatementSyntax } from 'glimmer-runtime';
+import { ConstReference, isConst, UNDEFINED_REFERENCE } from 'glimmer-reference';
+import { assert } from 'ember-metal/debug';
+import { RootReference } from '../utils/references';
+import { generateControllerFactory } from 'ember-routing/system/generate_controller';
+import { OutletLayoutCompiler } from './outlet';
+
+function makeComponentDefinition(vm) {
+  let env     = vm.env;
+  let args    = vm.getArgs();
+  let nameRef = args.positional.at(0);
+
+  assert(`The first argument of {{mount}} must be quoted, e.g. {{mount "chat-engine"}}.`, isConst(nameRef));
+
+  let name = nameRef.value();
+
+  assert(`You used \`{{mount '${name}'}}\`, but '${name}' can not be found as an engine.`, env.owner.hasRegistration(`engine:${name}`));
+
+  let engine = env.owner.buildChildEngineInstance(name);
+
+  let template = engine.lookup(`template:application`);
+
+  engine.boot();
+
+  return new ConstReference(new MountDefinition(name, engine, template, env, MOUNT_MANAGER));
+}
+
+export class MountSyntax extends StatementSyntax {
+  constructor({ args, symbolTable }) {
+    super();
+    this.definitionArgs = args;
+    this.definition = makeComponentDefinition;
+    this.args = ArgsSyntax.fromPositionalArgs(args.positional.slice(1, 2));
```

Comment:
I think this could just be `ArgsSyntax.empty()` – `{{mount}}` doesn't take more than one argument right?